src/Manifest/ManifestManager/Options/OutTable/ManifestOptions.php

```diff
@@ -0,0 +1,257 @@
+                    );
+                }
+            }
+        }
+
+        $this->primaryKeyColumns = $primaryKeyColumns;
+        return $this;
+    }
+
+    /**
+     * @param string[] $columns
+     */
+    public function setColumns(array $columns): ManifestOptions
+    {
+        if ($this->schema) {
+            throw new OptionsValidationException('Cannot set columns when schema is set');
+        }
+        $this->columns = $columns;
+        return $this;
+    }
+
+    public function addSchema(ManifestOptionsSchema $schema): ManifestOptions
+    {
+        if ($this->columns) {
+            throw new OptionsValidationException('Cannot set schema when columns are set');
+        }
+
+        if (!empty($this->primaryKeyColumns) && $schema->isPrimaryKeySet()) {
+            throw new OptionsValidationException(
+                'Only one of "primary_key" or "schema[].primary_key" can be defined.',
+            );
+        }
+
+        $this->schema[] = $schema;
+        return $this;
+    }
+
+    public function setIncremental(bool $incremental): ManifestOptions
+    {
+        $this->incremental = $incremental;
+        return $this;
+    }
+
+    public function setMetadata(array $metadata): ManifestOptions
+    {
+        if ($this->schema) {
+            throw new OptionsValidationException('Cannot set metadata when schema is set');
+        }
+        $this->validateMetadata($metadata);
+        $this->metadata = $metadata;
+        return $this;
+    }
+
+    /**
+     * @param mixed $columnsMetadata
+     */
+    public function setColumnMetadata($columnsMetadata): ManifestOptions
+    {
+        if ($this->schema) {
+            throw new OptionsValidationException('Cannot set column metadata when schema is set');
+        }
+        foreach ($columnsMetadata as $columnName => $columnMetadata) {
+            if (!is_array($columnMetadata)) {
+                throw new OptionsValidationException('Each column metadata item must be an array');
+            }
+            if (!is_string($columnName)) {
+                throw new OptionsValidationException('Each column metadata item must have string key');
+            }
+            try {
+                $this->validateMetadata($columnMetadata);
+            } catch (OptionsValidationException $e) {
+                throw new OptionsValidationException(sprintf('Column "%s": %s', $columnName, $e->getMessage()), 0, $e);
+            }
+        }
+        $this->columnMetadata = $columnsMetadata;
+        return $this;
+    }
+
+    public function setDelimiter(string $delimiter): ManifestOptions
+    {
+        $this->delimiter = $delimiter;
+        return $this;
+    }
+
+    public function setEnclosure(string $enclosure): ManifestOptions
+    {
+        $this->enclosure = $enclosure;
+        return $this;
+    }
+
+    public function setManifestType(string $manifestType): ManifestOptions
```

Comment:
Tady asi budou nějaký allowed values, který by se měli validovat, co? Nejsem si, ale jistej jaký. Nemůžu to na první dobrou najít v tom RFC.

Reply:
Je tam zatím jen hodnota "output", ale OJ ji teda nepřidal ani sem https://github.com/keboola/platform-libraries/blob/9ed89b21e0e3e8e27376595f4167d679e090f0e8/libs/output-mapping/src/Configuration/Table/Manifest.php#L17 což by se hodilo